api/src/main/scala/com/avast/clients/rabbitmq/api/RabbitMQManualConsumer.scala

```diff
@@ -0,0 +1,20 @@
+package com.avast.clients.rabbitmq.api
+
+import scala.language.higherKinds
+
+trait RabbitMQManualConsumer[F[_], A] {
+
+  /** Retrieves one message from the queue, if there is any.
+    *
+    * @return Some(DeliveryHandle[A]) when there was a message available; None otherwise.
+    */
+  def get(): F[Option[DeliveryWithHandle[F, A]]]
```

Comment:
Does it even make sense to return `None`? What I mean is that this API will be quite hard to use because you need to call `get` to get some wrapper (`Future`) which you need to add callback to and after that you need to check the `Option` just to find out that nothing is there... seems strange. I think in most cases you will want your thread to be blocked by no message available but I understand that you wanted to provide also non-blocking API. Do you have some concrete use case right now?

Reply:
The AMQP specs (http://www.rabbitmq.com/amqp-0-9-1-quickref.html#basic.get) says it returns a message or an _empty answer_. The Java RabbitMQ client respects that so it returns either the message or `null` - it does NOT wait for the message.

Reply:
Solved in person.